src/descriptor/mod.rs

```diff
@@ -848,7 +978,153 @@ impl<Pk: MiniscriptKey> fmt::Display for Descriptor<Pk> {
             Descriptor::Sh(ref sub) => write!(f, "sh({})", sub),
             Descriptor::Wsh(ref sub) => write!(f, "wsh({})", sub),
             Descriptor::ShWsh(ref sub) => write!(f, "sh(wsh({}))", sub),
+            Descriptor::ShSortedMulti(ref smv) => write!(f, "sh({})", smv),
+            Descriptor::WshSortedMulti(ref smv) => write!(f, "wsh({})", smv),
+            Descriptor::ShWshSortedMulti(ref smv) => write!(f, "sh(wsh({}))", smv),
+        }
+    }
+}
+
+/// Contents of a "sortedmulti" descriptor
+#[derive(Clone, PartialEq, Eq, PartialOrd, Ord)]
+pub struct SortedMultiVec<Pk: MiniscriptKey, Ctx: ScriptContext> {
+    /// signatures required
+    pub k: usize,
```

Comment:
I think we should use `u32` rather than `usize`. It's a bit less ergonomic but keeps us consistent across platforms, and we can't support such large numbers anyway.

Reply:
`Terminal::Multi` uses `usize`. Would it make sense to change that, too? 

It would be odd to use different types in these two places because they mean the same thing and we convert between them.

Reply:
Yeah, you're right, usize is fine